lib/grape/idempotency.rb

```diff
@@ -28,23 +28,28 @@ def idempotent(grape, &block)
         return block.call unless idempotency_key
 
         cached_request = get_from_cache(idempotency_key)
-        if cached_request && (cached_request["params"] != grape.request.params || cached_request["path"] != grape.request.path)
+        if cached_request && (cached_request["params"] != grape.request.params || cached_request["path"] != grape.request.path) && !cached_request["processing"]
```

Comment:
I'm not sure if this `!cached_request["processing"]` validation is needed. In this clause, we just want to check if there are conflicts, so I don't think we should take care of the processing status. 

Apart of that, the processing status will be checked in the next statement, so I'm not sure. What do you think?

Reply:
yes, not needed, as in the processing entry, there is no params so no conflict possibility